packages/components/src/components/drawer/drawer.jsx

```diff
@@ -15,87 +15,73 @@ const IconDrawer = ({ className }) => (
     </svg>
 );
 
-class Drawer extends React.PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = { is_open: props.is_open };
-    }
+const Drawer = ({
+    anchor = 'left',
+    children,
+    className,
+    contentClassName,
+    footer,
+    header,
+    width = 250,
+    zIndex = 4,
+    ...props
+}) => {
+    const [is_open, setIsOpen] = React.useState(props.is_open);
```

Comment:
Do we need this state? we can update props.is_open directly

Reply:
As I found in the code we may not pass the `toggleDrawer` function as a props to the `Drawer` component, so in this case we can not update `is_open` props directly and we should change it with local `toggleDrawer` method. so we can not remove the local method and we need `is_open` state as well.